#! /bin/sh /usr/share/dpatch/dpatch-run
## Fix FTBFS on alpha platforms.dpatch by Arthur Loiret <aloiret@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't abort compilation when alpha is detected, if linux is
## DP: detected as well. In this case, that's not Alpha OSF1, but
## DP: Linux on Alpha, which has dirent.h.

@DPATCH@

--- a/codec/dirent.h
+++ b/codec/dirent.h
@@ -118,7 +118,7 @@
 #   define HAVE_SYS_DIR_H
 # elif defined(__hpux)                         /* HP-UX */
 #   define HAVE_DIRENT_H
-# elif defined(__alpha) || defined(__alpha__)  /* Alpha OSF1 */
+# elif defined(__osf__) || defined(__digital__)  /* Alpha OSF1 */
 #   error "not implemented"
 # elif defined(__sgi)                          /* Silicon Graphics */
 #   define HAVE_DIRENT_Hdiff --git a/codec/dirent.h b/codec/dirent.h
